dictionary_amd64.go

```diff
@@ -69,12 +71,26 @@ func (d *doubleDictionary) lookup(indexes []int32, rows sparse.Array) {
 
 func (d *byteArrayDictionary) lookupString(indexes []int32, rows sparse.Array) {
 	checkLookupIndexBounds(indexes, rows)
-	dictionaryLookupByteArrayString(d.offsets, d.values, indexes, rows).check()
+	// TODO: this optimization is disabled for now because it appears to race
```

Comment:
Can we add the details of the CPU type that trigger this issue? Either in these comments or in the linked issue below?

That way if someone attempts a fix on this they can make sure to be testing on the CPU type that has a chance of reproing?